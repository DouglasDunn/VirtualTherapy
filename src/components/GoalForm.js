import React from 'react';

export default class GoalForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      goal: '',
      completed: false
    };
  }

  onGoalChange = (e) => {
    const goal = e.target.value;
    this.setState(() => ({ goal }));
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit({
      goal: this.state.goal,
      completed: this.state.completed
    });
    this.state.goal = '';
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="Goal"
          autoFocus
          className="text-input"
          value={this.state.goal}
          onChange={this.onGoalChange}
        />
        <div>
          <button className="button">Submit</button>
        </div>
      </form>
    )
  }
}
