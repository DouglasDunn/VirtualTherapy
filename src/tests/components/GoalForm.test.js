import React from 'react';
import { shallow } from 'enzyme';
import GoalForm from '../../components/GoalForm';
// import profile from '../fixtures/profile';

test('should render GoalForm correctly', () => {
  const wrapper = shallow(<GoalForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should set goal on input change', () => {
  const value = 'New goal';
  const wrapper = shallow(<GoalForm />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('goal')).toBe(value);
});
