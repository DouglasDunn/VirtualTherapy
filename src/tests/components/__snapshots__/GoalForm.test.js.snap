// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GoalForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Goal"
    type="text"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;
